xarray/core/dataarray.py

```diff
@@ -3234,6 +3237,128 @@ def map_blocks(
 
         return map_blocks(func, self, args, kwargs)
 
+    def polyfit(
+        self,
+        dim: Hashable,
+        deg: int,
+        skipna: bool = None,
+        rcond: float = None,
+        w: Union[Hashable, Any] = None,
+        full: bool = False,
+        cov: bool = False,
+    ):
+        """
+        Least squares polynomial fit.
+
+        This replicates the behaviour of `numpy.polyfit` but differs by skipping
+        invalid values when `skipna = True`.
+
+        Parameters
+        ----------
+        dim : hashable
+            Coordinate along which to fit the polynomials.
+        skipna : bool, optional
+            If True, removes all invalid values before fitting each 1D slices of the array
+            Defaults is True if data is stored in as dask.array or if there is any
+            invalid values, False otherwise.
+
+        Documentation of `numpy.polyfit` follows:
+        """
+        if skipna is None:
+            if isinstance(self.data, pycompat.dask_array_type):
+                skipna = True
+            else:
+                skipna = np.any(self.isnull())
+
+        x = get_clean_interp_index(self, dim)
+        order = int(deg) + 1
+        lhs = np.vander(x, order)
+
+        dims_to_stack = [dimname for dimname in self.dims if dimname != dim]
+        stacked_dim = utils.get_temp_dimname(dims_to_stack, "stacked")
+        rhs = self.transpose(dim, *dims_to_stack).stack({stacked_dim: dims_to_stack})
+
+        if rcond is None:
+            rcond = x.shape[0] * np.core.finfo(x.dtype).eps
+
+        # Weights:
+        if w is not None:
+            if isinstance(w, Hashable):
+                w = self[w]
+            w = np.asarray(w)
+            if w.ndim != 1:
+                raise TypeError("Expected a 1-d array for weights.")
+            if w.shape[0] != lhs.shape[0]:
+                raise TypeError("Expected w and {} to have the same length".format(dim))
+            lhs *= w[:, np.newaxis]
+            rhs *= w[:, np.newaxis]
+
+        # Scaling
+        scale = np.sqrt((lhs * lhs).sum(axis=0))
+        lhs /= scale
+
+        coeffs, residuals = duck_array_ops.least_squares(
+            lhs, rhs, rcond=rcond, skipna=skipna
+        )
+
+        if self.name:
+            name = "{}_".format(self.name)
```

Comment:
I think you'll end up with "name__polyfit_coefficient", is this intended ?

Reply:
No wasn't intended, you're right. Corrected in the next commit.